parlai/tasks/ms_marco/build.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+# Download and build the data if it does not exist.
+import gzip
+import json
+import os
+
+import parlai.core.build_data as build_data
+
+
+def read_gz(filename, delete_gz=True):
+    f = gzip.open(filename, 'rb')
+    lines = [x.decode('utf-8') for x in f.readlines()]
+    if delete_gz:
+        os.remove(filename)
+    return lines
+
+
+def create_fb_format(outpath, dtype, inpath):
+    print('building fbformat:' + dtype)
+
+    lines = read_gz(inpath)
+
+    # save the raw json version for span selection task (default)
+    fout1 = open(os.path.join(outpath, dtype + '.txt'), 'w')
+    for line in lines:
+        fout1.write(line.rstrip("\n") + "\n")
+    fout1.close()
+
+    # save the file for passage selection task
+    fout2 = open(os.path.join(outpath, dtype + '.passage.txt'), 'w')
+    for line in lines:
+        dic = json.loads(line)
+        lq = dic["query"]
+        if dtype != "test":
+            ans = "|".join([d["passage_text"] for d in dic["passages"] if d["is_selected"] == 1])
+            cands = "|".join([d["passage_text"] for d in dic["passages"] if d["is_selected"] == 0])
+            cands = ans + "|" + cands
+            if ans == "": continue  # if no true label, skip for now
+        else:  # ground truth for test data is not available yet
+            ans = ""
+            cands = "|".join([d["passage_text"] for d in dic["passages"]])
+        s = '1 ' + lq + '\t' + ans.lstrip("|") + '\t\t' + cands
+        fout2.write(s + '\n')
+    fout2.close()
+
+
+def build(opt):
+    dpath = os.path.join(opt['datapath'], 'MS_MARCO')
+
+    if not build_data.built(dpath):
```

Comment:
could you incorporate the changes to other build files here as well? we could also update it after landing this